Automattic/wp-calypso
# Data Stores: Fix persistence data might be incorrect

Related to #

## Proposed Changes

We use the `persistence` plugin to persist the store to the local storage on demand. However, if we register the plugin multiple times, the result will be incorrect and some of them will be overwritten with outdated data. Hence, this PR is proposing a change to wrap the registration in the single function to avoid the duplication.

## Testing Instructions



* Go to `/setup?siteSlug=<your_site>`
* Open the Devtool > Application > LocalStorage, and watch `WPCOM_7_DAYS_PERSISTENCE`
* When you back and forth between steps, the value of the `automattic/onboard` store should be updated.
  * For example, when you submit the selected goals, both goals, and intent will be updated to the selected values.
  * Note that the `verticalId` won't be updated since the vertical step doesn't set the value to the store. I tried to remove the `verticalId` from the store but looks like it's still used in the `storeProfiler` step

## Pre-merge Checklist
